src/parser/lexer.rs

```diff
@@ -386,6 +387,8 @@ pub enum LexicalError {
     InvalidAsciiEscapeCode(usize),
     /// Generic lexer error
     Generic(usize, usize),
+    /// Unbound type variable
+    UnboundTypeVariables(Vec<Ident>)
```

Comment:
An unbound type variable is not a lexical error. We should probably make a new custom error for LALRPOP, `CustomParseError` or whatever, that would wrap both lexical errors and unbound type variable errors.